File: backend/src/DueQ.Domain/Bills/Bill.cs
namespace DueQ.Domain.Bills;

public class Bill
{
    public Guid Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateOnly Date { get; set; }
    public string? Note { get; set; }
    public BillStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: frontend/projects/api/src/lib/models/bill.ts
import { BillStatus } from './bill-status';

export interface Bill {
  readonly id: string;
  readonly description: string;
  readonly amount: number;
  readonly partnerShare: number;
  readonly date: string;
  readonly note: string | null;
  readonly status: BillStatus;
  readonly createdAt: string;
}
